pkgs/applications/graphics/imgbrd-grabber/default.nix

```diff
@@ -48,6 +48,11 @@ stdenv.mkDerivation rec {
 
   extraOutputsToLink = [ "doc" ];
 
+  patches = [
+    # fix a compilation error that was added in a recent version of TypeScript
+    ./remove-unnecessary-conditional.patch
```

Comment:
It has been upstreamed, please `fetchpatch` this commit: https://github.com/Bionus/imgbrd-grabber/commit/21a06657831fa4026a5ac7bbd879dc7ac13cb278

Reply:
👍

Reply:
How can I make the patch be applied from the root of the source rather than inside the `src` directory? I avoided this before by editing the paths in the patch.

Example of what I mean:
```sh
# what it wants:
a/sites/Twitter/model.ts
# what it's getting:
a/src/sites/Twitter/model.ts
```

Reply:
This derivation sets `sourceRoot`, so you have to use `stripLen`:
```nix
  (fetchpatch {
    url = "https://github.com/Bionus/imgbrd-grabber/commit/21a06657831fa4026a5ac7bbd879dc7ac13cb278.patch";
    hash = "sha256-DAMk9Uvx7sqwTWLqb4Ghuh4JlLJloqrGnefMoVf5msQ=";
    stripLen = 1;
  })
```
And a comment above the patch explaining why it's needed.